### ISSUE TITLE
Make tutorial modal dialogs prettier

### ISSUE THREAD

**Issue by lazerwalker** [flappy-royale/flappy-royale#108]
Some beliefs / facts:

* We should make everybody pick a name. The random names are ugly, and are also long enough to break our designs.
* A playtester friend had never played Flappy Bird before, and was _super_ confused trying to figure out how to flap and move around for the first time while also not being able to tell who she was in the royale swarm
* I don't really want a hand-holding "you are a bird! tap to flap! don't hit the pipes!" style tutorial.

I'm not sure what this means for the correct onboarding experience. If we only care about fixing the name issue, we could also e.g. ask people for their name after they've had their first recorded game.

**Comment by orta** [comment]
our extra elements don't help:

![Screen Shot 2019-06-22 at 8 52 12 AM](https://user-images.githubusercontent.com/49038/59966028-0a160d00-94cb-11e9-8836-fac78540ddd1.png)

**Comment by lazerwalker** [comment]
haha, yeahhh.

A thought at a more involved tutorial that doesn't feel too infantalizing:

The game starts with a bus with just you in it, flying infinitely (no pipes) until you tap. There's a 'tap the screen' icon mid-screen, like in the original Flappy Bird onboarding.

When you tap, you exit the bus like normal. There are still no pipes. If you hit the ground, we reset immediately to the infinite bus.

A bit ahead of you (so, far enough you have to flap a bunch) there's a whole bunch of bottom pipes right next to each other (so, like a platform) with a whole bunch of birds resting on them. Maybe there are some top pipes, but the bottom ones are low enough and the top ones are high enough that it's not designed to be a challenge.

As soon as you clear the first pipe, we pop a dialog box with a text box saying something like "Welcome to the Royale! What's your name?". After that, we kick you to the main menu.

It teaches everybody how the bus works, experienced Flappy Bird players can skip past it in a matter of seconds, and it makes sure that totally new players at least know how to drop from the bus and then flap.

**Comment by orta** [comment]
Yep

**Comment by orta** [comment]
Think this should also have a button on the main screen or settings too

**Comment by orta** [comment]
because I hand this to people all the time and would love them to go through it

**Comment by lazerwalker** [comment]
Yep, think there could be a "how to play" button on the main menu. v2 of this could be that explicit tutorial version having another "avoid the pipes to score points" step to really drill home the rules.

Also making a note to myself to confirm that the "reset game" settings menu option also forces you to play the tutorial again.

**Comment by lazerwalker** [comment]
FYI, a first stab at this is messily in-progress. Probably halfway there.

**Comment by lazerwalker** [comment]
I did a tiiiny bit more work on this and pushed/deployed. It requires an iOS client redeploy as well. (Haven't tested Android)

It technically works, but lacks polish. I figured shipping this and then getting feedback made sense.

Current things that I believe need work:

- [ ] Add a 'tap the screen' icon while in the bus / flapping
- [ ] Have a bunch of birds with random attire hanging out on the pipes (stretch goal: they randomly jump around a little bit)
- [ ] Add a "Royale ->" hanging sign just before the pipes
- [ ] Play the 'bus taking off' sound effect at the start of the tutorial
- [ ] The "what's your name?" prompt is a modal Phaser scene. Gray background, random color blobs for flavor, an HTML text input form
- [ ] The "go to settings?" prompt is also a modal Phaser scene
- [ ] Give the main menu "?" button a purple background (optional stretch goal: make a dedicated English-text "how to play" button? I think this is fine)
- [ ] Let both `prompt()` and `alert()` work on Android (low-priority, since it'll likely get replaced by HTML)
- [ ] What happens if you click 'cancel' in the name prompt? (low-priority, since it'll likely get replaced by HTML)

**Comment by lazerwalker** [comment]
Thinking through what a subtler version of this is. My current thoughts:

- When we load the main menu, if you don't have a name, instead show a "pick a name" screen, followed by a "pick some clothes?" prompt.

- The first game you play (first X games?) we secretly drop the alpha on the ghost birds, so it's easier to figure out who you are. If we end up having a "tap the screen to flap" image, maybe we always show it on-screen until you leave the bus?

- Either as part of the name flow, or when you first tap a game mode, we explicitly ask you if you've played Flappy Bird before. If you say no, we take you to an onboarding that's similar to the one I built yesterday, but maybe with more explicit text.

Gonna think through this some more.

**Comment by orta** [comment]
Nice, these are all great ideas

**Comment by lazerwalker** [comment]
Currently feeling like a forced "pick your name" flow is a blocker for public beta, whereas the other two are slightly less so.

**Comment by lazerwalker** [comment]
Name + attire flow is in place, although I intend to replace the solid purple box with a jauntier angled box (went to go buy Asesprite to make it myself, but lol, got myself flagged on Humble for a sketchy purchase)

Needs #140 to be really usable.

<img width="538" alt="Screen Shot 2019-06-25 at 13 11 44" src="https://user-images.githubusercontent.com/821238/60093876-d517ed80-974a-11e9-82c5-25ddcb2e5bb1.png">

<img width="538" alt="Screen Shot 2019-06-25 at 13 11 47" src="https://user-images.githubusercontent.com/821238/60093863-cc271c00-974a-11e9-99dc-b7b9e022c896.png">

**Comment by orta** [comment]
I use [Pixen](https://pixenapp.com) it's buggy but good enough - I can help make the font rendering sharp too

I think it might be worth making the tutorial a little longer:
- maybe only first having bottom pipes
- then moving to top + bottom to really hammer in the getting used to it

I wonder if it it could do with some kind of finishing indication, I initially thought it was a bug that it was asking me questions. Maybe I can whip up a finishing flag?

**Comment by lazerwalker** [comment]
Yeah, if you want to whip a flag, that sounds awesome!

Otherwise, I'm in the middle of some changes to the tutorial that'll hoooopefully make it a bit more obvious.

**Comment by lazerwalker** [comment]
Okay, deploying a draft of the new tutorial. I _believe_ this will force a popup on all existing users the next time they open the game, but such is life. You can force the tutorial to pop by clicking the question mark as well.

(The name + attire prompts are triggered regardless of tutorial status — you can force them to show by going into settings and erasing your name, then going back)

It still ends more abruptly than I'd like, which is where I think reaching a flag would help a lot. Otherwise, would love any other feedback!

**Comment by lazerwalker** [comment]
I suspect this still may be a brutal introduction for people who don't know what they're doing, but given both Flappy Bird and battle royales I'm also okay with only giving people only one or two nudges more than "shrug, figure it out".

**Comment by orta** [comment]
<img width="897" alt="Screen Shot 2019-06-25 at 9 53 10 PM" src="https://user-images.githubusercontent.com/49038/60151975-a4a87180-9793-11e9-8370-ff2bb9f85fc7.png">

**Comment by orta** [comment]
Flap in Fashion is great copy

**Comment by orta** [comment]
OK, I've made a tweak or two, when you to the flag the "computer" takes over flapping for you and the bird starts moving offscreen so that it won't just die on the floor

**Comment by orta** [comment]
Flow wise, yep, this all feels good to me. I've nuked my local storage a few ties to go through the full flow.

I think we need to make these alerts a tad more in yer face - they need some janky angles, and some random shapes probably, maybe the yellow ones for the thin one, and some of the sharp reds maybe for the ones with buttons. 

Not a blocker, and if you leave it, I'll do it eventually and try get the text sharper

**Comment by lazerwalker** [comment]
Coool, agree completely, just wanted to see if the flow felt right before polishing.

**Comment by lazerwalker** [comment]
(slight awkwardness around the tutorial ones is we can't really move them any higher up — since they're their own scene, there's no way to set their z-depth as behind the player bird but in front of everything else. Hence me shoving 'em at the bottom of the screen)
